src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double INCREASE_RATE = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[CAPACITY];
+    }
+
+    public void increaseArray() {
+        T[] arrayCopy = (T[]) new Object[(int) (array.length * INCREASE_RATE)];
+        System.arraycopy(array, 0, arrayCopy, 0, array.length);
+        array = arrayCopy;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() >= array.length) {
+            increaseArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
```

Comment:
Please change 'size()' to 'size' everywhere. It's easier to read the code when we see call to the variable, but not to method
```suggestion
        if (index < 0 || index > size) {
```